src/Console/BaseCommand.php

```diff
@@ -128,7 +138,7 @@ public function getOptionParser(): ConsoleOptionParser
      */
     protected function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionParser
     {
-        return $parser;
+        return $parser->setDescription(static::getDescription());
```

Comment:
This method is usually fully overridden, so description will not be set when overridden.

I think it is better to set description in `getOptionParser()` before calling `buildOptionParser()` method.

Reply:
I thought due to the fact that the parent method is called by default (at least when baking the command) this would be fine.
https://github.com/cakephp/bake/blob/2.x/templates/bake/Command/command.twig#L40

Reply:
All core [commands](https://github.com/cakephp/cakephp/tree/4.x/src/Command), samples in [docs](https://book.cakephp.org/4/en/console-commands/commands.html) doesn't call parent.

Reply:
well not all, but certainly most of them 😉 
Anyway I will adjust that to the `getOptionParser()`
<img width="787" alt="image" src="https://user-images.githubusercontent.com/9105243/162518934-08f7eac5-c33b-4279-b482-241dd4198428.png">